src/image/image-proc.ts

```diff
@@ -0,0 +1,159 @@
+import Jimp from 'jimp';
+import { img2array, tensor2Img, Tensor2ImgArgs, std } from './utils';
+import * as tf from '@tensorflow/tfjs-core';
+import '@tensorflow/tfjs-backend-cpu';
+
+/**
+ * Image class contains utility to handle image manipulation
+ * @contructor (name: string) -> name is the image file name.
+ */
+export default class Image {
+  private img: Jimp; //img accessible to all methods in the class
+
+  constructor (data: Jimp){
+    this.img = data;
+  }
+
+  /**
+   * Read image from files and create an Image object for processing
+   * @param name image file name
+   */
+  static async read(name: string): Promise<Image>{
+
+    const jimp_obj = await Jimp.read(name);
+
+    return new Image(jimp_obj);
+  }
+
+  /**
+   * Convert image Buffer to Tensor
+   * @param depth -> number of channel to obtain from image
+   * @return Promise<Array<Array<Array<number>>>> -> A 3dimensional array
+   */
+  public img2tensor(depth = 3): tf.Tensor3D{
```

Comment:
```suggestion
  public toTensor(depth = 3): tf.Tensor3D {
```

The image should be in the context.